src/VizTab.tsx

```diff
@@ -74,18 +208,19 @@ export default function VizTab(props: {namespace: string}) {
     </List>
   </SettingsPanel>;
 
-  // just extract the controls to here and pass in props
-
+  // these viewers will be rendered into the <Canvas>
   // since <Canvas> uses the react-three-fiber reconciler, we must forward
   // any contexts manually :(
   const viewers = <WebSocketContext.Provider value={ws}>
-      <Localization2DViewer
-        namespace={props.namespace}
-        topic="localization"
-        mapColor={0x536dfe}
-        mapVisible={locShowMap}
-        poseColor={0x8ECC47}
-      />
+      <RobotContext.Provider value={rc}>
+        <Localization2DViewer
+          namespace={props.namespace}
+          topic="localization"
```

Comment:
if you need to drill down deeper than one component, I'm all for contexts. Here, might not need one if you pull the state up to VizTab and pass a prop instead